auth/service.go

```diff
@@ -0,0 +1,59 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	authclient "github.com/fabric8-services/fabric8-auth-client/auth"
+	"github.com/fabric8-services/fabric8-common/errors"
+	"github.com/fabric8-services/fabric8-common/goasupport"
+
+	goaclient "github.com/goadesign/goa/client"
+)
+
+type AuthService interface {
+	RequireScope(ctx context.Context, resourceID, requiredScope string) error
+}
+
+func NewAuthService(authURL string) (AuthService, error) {
+	u, err := url.Parse(authURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return &auth{u}, nil
+}
+
+type auth struct {
+	authURL *url.URL
+}
+
+func (a *auth) RequireScope(ctx context.Context, resourceID, requiredScope string) error {
+	client := a.createClient(ctx)
+	resp, err := client.ScopesResource(goasupport.ForwardContextRequestID(ctx), authclient.ScopesResourcePath(resourceID))
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return errors.NewInternalErrorFromString(fmt.Sprintf("get space's scope failed with error '%s'", resp.Status))
+	}
+
+	defer resp.Body.Close()
+	scopes, _ := client.DecodeResourceScopesData(resp)
+	for _, scope := range scopes.Data {
+		if requiredScope == scope.ID {
+			return nil
+		}
+	}
+	return errors.NewForbiddenError(fmt.Sprintf("missing required scope '%s' on '%s' resource", requiredScope, resourceID))
+}
+
+func (a *auth) createClient(ctx context.Context) *authclient.Client {
```

Comment:
Comment here also please

Reply:
Done in https://github.com/fabric8-services/fabric8-common/pull/62/commits/1076d9185c0ec1a4d3d2e65c316aa65346099d9a